src/main/java/bio/terra/dao/ProfileDao.java

```diff
@@ -0,0 +1,85 @@
+package bio.terra.dao;
+
+import bio.terra.dao.exception.AccountAlreadyExistsException;
+import bio.terra.dao.exception.ProfileNotFoundException;
+import bio.terra.metadata.BillingProfile;
+import bio.terra.metadata.MetadataEnumeration;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.UUID;
+
+@Repository
+public class ProfileDao {
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+
+    @Autowired
+    public ProfileDao(NamedParameterJdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    public UUID createBillingProfile(BillingProfile billingProfile) {
+        try {
+            String sql = "INSERT INTO billing_profile (name, biller, billing_account_id) VALUES " +
+                "(:name, :biller, :billing_account_id)";
+            MapSqlParameterSource params = new MapSqlParameterSource()
+                .addValue("name", billingProfile.getName())
+                .addValue("biller", billingProfile.getBiller())
+                .addValue("billing_account_id", billingProfile.getBillingAccountId());
+            DaoKeyHolder keyHolder = new DaoKeyHolder();
+            jdbcTemplate.update(sql, params, keyHolder);
+            return keyHolder.getId();
+        } catch (DuplicateKeyException ex) {
+            throw new AccountAlreadyExistsException("Account in use: " + billingProfile.getBillingAccountId(), ex);
```

Comment:
the name of this exception is misleading. the billing account should exist. it just shouldn't already be used for a billing profile. maybe something like AccountAlreadyInUse. also if it doesn't break security, it would be nice to return the billing profile that is already using that account. can be done later

Reply:
Great point, I'm going to update to `AccountAlreadyInUse`. I'll also add the profile id that is using the account -- I think it's a good pattern.